### PR TITLE
My artists rest controller

### PR DESCRIPTION
**Pull request by hedgehog1833**
- MyArtistRestController implemented
- Added Pageable also to ArtistRestController
- ArtistService extended

### CODE REVIEW

**Comment by hedgehog1833** [comment]
[Create RestController for MyArtists](https://trello.com/c/IFMS7JI9/48-create-restcontroller-for-myartists)

File: src/test/java/com/metalr2/web/controller/rest/MyArtistsRestControllerIT.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        .contentType(ContentType.JSON)
+        .statusCode(HttpStatus.OK.value());
+
+    MyArtistsResponse response = validatableResponse.extract().as(MyArtistsResponse.class);
+
+    assertThat(response).isNotNull();
+    assertThat(response.getMyArtists()).hasSize(1);
+    assertThat(response.getMyArtists().get(0).getDiscogsId()).isEqualTo(DISCOGS_ID);
+    assertThat(response.getMyArtists().get(0).getArtistName()).isEqualTo(ARTIST_NAME);
+    assertThat(response.getPagination()).isNotNull();
```
Comment (DanielW1987) [inline src/test/java/com/metalr2/web/controller/rest/MyArtistsRestControllerIT.java]: This is the only place where I would test Pagination...Fits the response to my request.

File: src/main/java/com/metalr2/model/artist/FollowedArtistsRepository.java [hunk 2]
```diff
@@ -11,8 +12,12 @@
 
   List<FollowedArtistEntity> findAllByPublicUserId(String publicUserId);
 
-  boolean existsByPublicUserIdAndArtistDiscogsId(String publicUserId, long artistDiscogsId);
+  boolean existsByPublicUserIdAndDiscogsId(String publicUserId, long discogsId);
 
-  Optional<FollowedArtistEntity> findByPublicUserIdAndArtistDiscogsId(String publicUserId, long artistDiscogsId);
+  Optional<FollowedArtistEntity> findByPublicUserIdAndDiscogsId(String publicUserId, long discogsId);
+
+  List<FollowedArtistEntity> findAllByPublicUserIdAndDiscogsIdIn(String publicUserId, long... discgogsIds);
```
[the remark this answers is not in the document]
Comment (hedgehog1833) [reply src/main/java/com/metalr2/model/artist/FollowedArtistsRepository.java]: Hmm I'm also not sure anymore why I thought I needed it.... I'll delete it for now

File: src/test/java/com/metalr2/model/artist/ArtistsRepositoryTest.java [hunk 3]
```diff
@@ -73,12 +76,40 @@ void exists_by_artist_discogs_id_should_return_false() {
...
+      assertThat(entity.getThumb()).isNull();
+    }
+  }
+
+  private static Stream<Arguments> inputProviderDiscogsIds() {
+    return Stream.of(
+        Arguments.of(Arrays.asList(2L, 1L)),
+        Arguments.of(Collections.singletonList(1L)),
+        Arguments.of(Collections.emptyList())
+    );
```
[the remark this answers is not in the document]
Comment (hedgehog1833) [reply src/test/java/com/metalr2/model/artist/ArtistsRepositoryTest.java]: I tried it that way first but parametrized tests cannot take varargs. But I just found out that an array works, when it is cast to object in the .of() method.

File: src/test/java/com/metalr2/service/artist/ArtistsServiceTest.java [hunk 4]
```diff
@@ -333,40 +341,138 @@ void is_followed_should_return_false_for_not_existing_entity(){
...
+      Pagination pagination = myArtists.getPagination();
+
+      assertThat(pagination).isNotNull();
+      assertThat(pagination.getCurrentPage()).isEqualTo(pageRequest.getPageNumber());
+      assertThat(pagination.getItemsPerPage()).isEqualTo(pageRequest.getPageSize());
+      assertThat(pagination.getTotalPages()).isEqualTo(totalPagesExpected);
+
+      verify(followedArtistsRepository, times(1)).findAllByPublicUserId(USER_ID, pageRequest);
+      verify(artistsRepository, times(1)).findAllByArtistDiscogsIdIn(any());
+    }
```
[the remark this answers is not in the document]
Comment (hedgehog1833) [reply src/test/java/com/metalr2/service/artist/ArtistsServiceTest.java]: Yeah this one does not make this much sense anymore anyway as the pagination object is not built there anymore.

File: src/test/java/com/metalr2/model/artist/ArtistsRepositoryTest.java [hunk 5]
```diff
@@ -6,44 +6,46 @@
...
+import static com.metalr2.web.DtoFactory.ArtistEntityFactory;
 
 @DataJpaTest
 class ArtistsRepositoryTest implements WithAssertions, WithIntegrationTestProfile {
 
-  private static final long artistDiscogsId1 = 245797;
-  private static final long artistDiscogsId2 = 252211;
-  private static final String artistName1 = "Opeth";
-  private static final String artistName2 = "Darkthrone";
-  private static final String thumb = "thumb";
```
[the remark this answers is not in the document]
Comment (hedgehog1833) [reply src/test/java/com/metalr2/model/artist/ArtistsRepositoryTest.java]: I made some adjustments and used constants where I found it sensible. The test classes now controll the naming of the DTOs.

File: src/main/java/com/metalr2/web/controller/rest/MyArtistsRestController.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  @Autowired
+  public MyArtistsRestController(ArtistsService artistsService) {
+    this.artistsService = artistsService;
+  }
+
+  @GetMapping(produces = {MediaType.APPLICATION_XML_VALUE, MediaType.APPLICATION_JSON_VALUE})
+  public ResponseEntity<MyArtistsResponse> getMyArtists(@PageableDefault Pageable pageable) {
+    List<ArtistDto> artists = artistsService.findFollowedArtistsForCurrentUser(pageable);
+    long totalArtists = artistsService.countFollowedArtistsForCurrentUser();
+    Pagination pagination = new Pagination((int) Math.ceil((double) totalArtists / pageable.getPageSize()),
```
Comment (DanielW1987) [inline src/main/java/com/metalr2/web/controller/rest/MyArtistsRestController.java]: With a later PR you can implement that the total pages are calculated within the Pagination class. You pass only `total`, `page` and `size` to the constructor.

**Pull request merged**